tree/ntuple/v7/test/ntuple_types.cxx

```diff
@@ -2233,3 +2229,30 @@ TEST(RNTuple, RColumnRepresentations)
                                                   {ROOT::ENTupleColumnType::kReal16}}),
              colReps2.GetDeserializationTypes());
 }
+
+TEST(RNTuple, RContextDependentTypes)
+{
+   // Adapted from https://gitlab.cern.ch/amete/rntuple-bug-report-20250219, reproducer of
+   // https://github.com/root-project/root/issues/17774
+
+   FileRaii fileGuard("test_ntuple_types_contextdep.root");
+
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldBase = RFieldBase::Create("foo", "DerivedWithTypedef").Unwrap();
+      model->AddField(std::move(fieldBase));
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      for (auto i = 0; i < 10; ++i) {
+         auto entry = ntuple->GetModel().CreateBareEntry();
+         auto ptr = std::make_unique<DerivedWithTypedef>();
+         ptr->m.push_back(i);
+         entry->BindRawPtr("foo", ptr.get());
+         ntuple->Fill(*entry);
+      }
```

Comment:
Random question.  Is this code also valid (and lead to the same output)?:
```suggestion
      auto entry = ntuple->GetModel().CreateBareEntry();
      auto ptr = std::make_unique<DerivedWithTypedef>();
      entry->BindRawPtr("foo", ptr.get());
      for (auto i = 0; i < 10; ++i) {
         ptr->m.clear();
         ptr->m.push_back(i);
         ntuple->Fill(*entry);
      }
```

Reply:
Yes, entries can be reused and `BindRawPtr` called infinitely often. In the end it's just a container / staging area for one call to `Fill()`.